scanner/indexer/indexer.go

```diff
@@ -255,3 +255,36 @@ func parseContainerImageURL(imageURL string) (name.Reference, error) {
 	}
 	return ref, nil
 }
+
+// GetDigestFromURL returns an image digest from the given image URL.
+func GetDigestFromURL(imgURL string, auth authn.Authenticator) (name.Digest, error) {
```

Comment:
I don't think these two functions are necessarily tied to the `indexer` (they are used by `scannerctl`, for example). I think perhaps a new package may be useful for these. Something like `digest` (or just putting it in `scannerctl`, as I don't think these are used elsewhere at the moment?)